src/services/dates.js

```diff
@@ -0,0 +1,67 @@
+export const weekdayShortNames = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];
+
+function getWeekdayName(dayId) {
+  const errorMessage = `
+  No matching ID to weekday. 
+  Used ID: ${dayId}, while possible IDs are from 0 to 6.
+  `;
+  if (dayId === 0) return 'Sunday';
+  if (dayId === 1) return 'Monday';
+  if (dayId === 2) return 'Tuesday';
+  if (dayId === 3) return 'Wednesday';
+  if (dayId === 4) return 'Thursday';
+  if (dayId === 5) return 'Friday';
+  if (dayId === 6) return 'Saturday';
+  else throw new Error(errorMessage);
+}
+
+function padStartFormat(number) {
+  return number.toString().padStart(2, '0');
+}
+
+function getApiDateFormat(year, month, day) {
+  return `${year}-${padStartFormat(month)}-${padStartFormat(day)}`;
+}
+
+function getDisplayDateFormat(year, month, day) {
+  return `${padStartFormat(day)}/${padStartFormat(month)}/${year}`;
+}
+
+export function getFormattedDateObject(newDateInstance) {
+  const year = newDateInstance.getFullYear();
+  const month = newDateInstance.getMonth();
+  const actualMonth = month + 1;
+  const day = newDateInstance.getDate();
+  const weekdayIndex = newDateInstance.getDay();
+  return {
+    meta: {
+      year,
+      month,
+      day,
+    },
+    apiDate: getApiDateFormat(year, actualMonth, day),
+    displayDate: getDisplayDateFormat(year, actualMonth, day),
+    weekday: getWeekdayName(weekdayIndex),
+    shortWeekday: getWeekdayName(weekdayIndex).slice(0, 3),
+  };
+}
+
+function getFollowingDays(todaysYear, todaysMonth, todaysDay) {
+  const followingDays = [];
+  for (let i = 1; i <= 5; i++) {
+    const followingDayInstance = new Date(todaysYear, todaysMonth, todaysDay + i);
```

Comment:
What will happen if todaysDay is 30?

Reply:
It will go to the next month or to 31st day of the month depending on what is the current month.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#syntax
> If any parameter overflows its defined bounds, it "carries over". For example, if a monthIndex greater than 11 is passed in, those months will cause the year to increment; if a minutes greater than 59 is passed in, hours will increment accordingly, etc. Therefore, new Date(1990, 12, 1) will return January 1st, 1991; new Date(2020, 5, 19, 25, 65) will return 2:05 A.M. June 20th, 2020.

Reply:
Ok nice 🙂  👍